pybuildkite/client.py

```diff
@@ -49,15 +50,19 @@ def request(
         :param with_pagination: Bool to return a response with pagination attributes
         :return: If headers are set response text is returned, otherwise parsed response is returned
         """
+        if headers is None:
+            raise ValueError("headers cannot be None")
 
-        query_params = self._clean_query_params(query_params or {})
+        if not headers.get("Accept"):
```

Comment:
We don't need to force `headers["Accept"] = "application/json"`, I think we can allow for `Accept` to not exist as a key.